### PR TITLE
[Mono.Posix] Fix the Android xattr check to not disable it on all other platforms.

### PR DESCRIPTION
**Pull request by kumpera**
Fixes https://github.com/mono/mono/issues/6794

### CODE REVIEW

**Comment by kumpera** [comment]
@monojenkins backport to 2017-10

**Comment by kumpera** [comment]
@monojenkins backport to 2017-12

File: support/sys-xattr.c [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include <config.h>
 
 //If we're compiling to API level < 16 this won't be available
-#if __ANDROID_API__ < 16
+#if defined (HOST_ANDROID) && __ANDROID_API__ < 16
```
Comment (vargaz) [inline support/sys-xattr.c]: Why does this #if succeed on non-android platforms ?

Comment (jaykrell) [reply support/sys-xattr.c]: undefined evaluates as 0, silently, per the language. Microsoft has a warning for this but it isn't enabled by default and really it is just expected behavior and a warning is dubious

I suggest similar:

jays-MacBook-Pro:support jay$ git diff -u
+#ifdef __ANDROID_API__
 #if __ANDROID_API__ < 16
 #define ANDROID_NO_XATTR
 #endif
+#endif

**Comment by lewurm** [comment]
the regression was introduced with https://github.com/mono/mono/commit/c133304b9510b6efcbe013d5dbcc73cbf336e991 which didn't make it into `2017-10`, so it only needs to be backported to `2017-12` (and `2018-02` since today 🙂  )

**Comment by marek-safar** [comment]
build

**Comment by luhenry** [comment]
@kumpera the backport to 2017-10 ended up failing because of conflict, is that a problem? Should we backport manually?

**Comment by akoeplinger** [comment]
it's also missing in 2018-02

**Comment by akoeplinger** [comment]
@luhenry it shouldn't be necessary in 2017-10 according to https://github.com/mono/mono/pull/6798#issuecomment-363203874

@monojenkins backport 2018-02

**Pull request merged**